File: Go/BasePriciples/Law_of_Demeter/GoVersion/Agent.go
package LODGoversion

import "fmt"

type Agent struct{
	fans Fans
	start Start
	company Company
}

func (a *Agent) SetFans(fans Fans) {
	a.fans = fans
} 

func (a *Agent) SetStart(start Start) {
	a.start = start
}

func (a *Agent) SetCompany(company Company) {
	a.company = company
} 

func (a Agent) Meeting() {
	fmt.Println(a.start.getName()+"和粉丝"+a.fans.getName()+"见面")
}

func (a Agent) Business() {
	fmt.Println(a.start.getName()+"和"+a.company.getName()+"洽谈")
}

File: JavaVersion/src/com/basePriciples/Law_of_Demeter/Agent.java
package com.basePriciples.Law_of_Demeter;

public class Agent {
    private Start start;
    private Fans fans;
    private Company company;
    public void setStart(Start start) {
        this.start = start;
    }
    public void setFans(Fans fans) {
        this.fans = fans;
    }
    public void setCompany(Company company) {
        this.company = company;
    }

    public void meeting() {
        System.out.println(start.getName()+"和粉丝"+fans.getName()+"见面");
    }

    public void business() {
        System.out.println(start.getName()+"和"+company.getName()+"洽谈");
    }
}
